recipes/apr-util/all/conanfile.py

```diff
@@ -45,162 +55,165 @@ class AprUtilConan(ConanFile):
         "with_ldap": False,
     }
 
-    _autotools = None
-    _cmake = None
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+        self.options["apr"].shared = self.options.shared
 
-        if not self.options.with_expat:
-            raise ConanInvalidConfiguration("expat cannot be disabled (at this time) (check back later)")
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def layout(self):
+        if self.settings.os == "Windows":
+            cmake_layout(self, src_folder="src")
+        else:
+            basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("apr/1.7.0")
+        self.requires("apr/1.7.0", transitive_headers=True)
         if self.settings.os != "Windows":
             #cmake build doesn't allow injection of iconv yet
-            self.requires("libiconv/1.16")
+            # https://github.com/conan-io/conan-center-index/pull/16142#issuecomment-1494282164
+            self.requires("libiconv/1.17", transitive_libs=True)
```

Comment:
@prince-chrismc It makes little sense. If it's a weird workaround for macOS agents of c3i, it should be documented.

Reply:
🤔 I am pretty sure this is the same problem from https://github.com/conan-io/conan-center-index/pull/16266/files#diff-bbf22e1d844ff0998d0eeeb0debd0266d654ed0d1e46716a45f311fcde9760e1R107 and the team is working on fixit it

I will improve the error mesaage though :)

Reply:
```suggestion
            # https://github.com/conan-io/conan-center-index/pull/16142#issuecomment-1494282164
            # transitive_libs needs to be set because some sys-frameworks on the old mac images for c3i
            # are pulling it in - discovered in https://github.com/conan-io/conan-center-index/pull/16266
            self.requires("libiconv/1.17", transitive_libs=True)
```